### ISSUE TITLE
Unencapsulated YBR_CRT photometric interpretation is failing 

### ISSUE THREAD

**Issue by EdwardMarcus** [fo-dicom/fo-dicom#1437]
### Expected behavior

I made this standard call to the pixel data factory 
            'returns from non encapsulated pixeldata
            Return Imaging.Render.PixelDataFactory.Create(PixelData, CurrentFrame) 

### Actual behavior

Call to the pixel data factory fails  when photometric interpretation is ybr-crt.

### Steps to reproduce the behavior
Call to the pixel data factory fails  when photometric interpretation is ybr-crt.

### fo-dicom version and OS/platform
5.0 win 7 pro

Not sure if this is a bug, or might require some pixel manipulations?

Thank you

Ed

**Comment by EdwardMarcus** [comment]
When encountering YBRRCT I have implemented a swap to RGB
  If PixelData.PhotometricInterpretation.Equals(PhotometricInterpretation.YbrRct) Then PixelData.PhotometricInterpretation = 
PhotometricInterpretation.Rgb

 Return Imaging.Render.PixelDataFactory.Create(PixelData, CurrentFrame) 

I now get the render-- albeit with distorted colors. 

I guess I will need to try an YBRRCT=>RGB color space conversion.
Thanks
Ed

**Comment by mrbean-bremen** [comment]
Looks like support for YBRLCT and YBRRRCT is not implemented yet.
In `DicomImage.CreatePipelineData` there is this code:
```cs
            // temporary fix for JPEG 2000 Lossy images
            if (pi == PhotometricInterpretation.YbrIct || pi == PhotometricInterpretation.YbrRct)
            {
                pi = PhotometricInterpretation.Rgb;
            }
```
which just prevents a crash, but does no color space correction. 
In `PixelDataFactory.Create` these cases are not handled at all, which will cause a `DicomImagingException`.

As you noticed, we will need to add the color space conversion (similar to YBR to RGB conversion, though I don't know about these color spaces), and handle this case accordingly.

**Comment by EdwardMarcus** [comment]
I tried this Y,CB,CR => R,G,B color convert.

https://fo-dicom.github.io/html/4c7bc42c-e059-361e-81e2-b6a4bbd384e1.htm

In some instances  R, G, or B  is exceeding 255. The need to compute math.floor is very slow.

So I am probably missing something.

**Comment by gofal** [comment]
@EdwardMarcus with the PR #1591 a lot of rendering-related issues have been fixed. Also the YBRLCT should be supported now.
Could you please give it a try with the latest version in development-branch of fo-dicom, if this issue is still open?

**Comment by EdwardMarcus** [comment]
Reinhard,

I am still working on version 4 in my production software, and I need to update to the latest version 5 eventually.

Maybe I can sandbox a transfer syntax change to jpeg baseline in V5, and I can let you know what I see.

Thank you!

Ed

**Comment by gofal** [comment]
Closing due to inactivity. If you have tried with latest version of fo-dicom and still find the problem, then please reopen this issue and provide actual informations.

**Comment by EdwardMarcus** [comment]
Hello Reinhard,

I apologize for not checking out the modified feature in V 5.0. ( I am implemented in V 4.0, and I will eventually need to make a transition to 5.0)

Thank you for staying on top of the issue nevertheless.

Kindly reach out if interested in developing DICOM implementations.

My best,

Ed Marcus


Principal Research Software Engineer
Cardiology, Boston Children’s Hospital

***@***.***
